ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/SubmitterInfo.scala

```diff
@@ -25,10 +27,20 @@ import com.digitalasset.daml.lf.data.Time.Timestamp
   *   by DAML applications to deduce from the record time reported by the
   *   ledger whether a change that they submitted has been lost in transit.
   *
+  * @param deduplicateUntil: the time until which the command should be
+  *   deduplicated. Command deduplication is already performed by the
+  *   participant. The ledger may choose to perform additional
+  *   (cross-participant) command deduplication.
+  *   If it chooses to do so, it must follow the
+  *   same rules as the participant:
+  *   - Deduplication is based on the (submitter, commandId) tuple.
+  *   - Commands must not be deduplicated after the TTL has passed.
+  *   - Commands should not be deduplicated after the command was rejected.
   */
 final case class SubmitterInfo(
     submitter: Party,
     applicationId: ApplicationId,
     commandId: CommandId,
-    maxRecordTime: Timestamp //TODO: this should be a regular Instant
+    maxRecordTime: Timestamp, //TODO: this should be a regular Instant
+    deduplicateUntil: Instant,
```

Comment:
We now have two terms for the TTL:
1. `deduplicateUntil`: absolute time
1. `maxCommandTtl`: duration

I suggest that we unify those so that it's less confusing when talking about it. Maybe we switch to absolute time on the ledger api as well, using `deduplicate_until`?